cmd/flux/main.go

```diff
@@ -138,30 +141,17 @@ func NewRootFlags() rootFlags {
 
 func main() {
 	log.SetFlags(0)
-	configureKubeconfig()
-	configureDefaultNamespace()
 	if err := rootCmd.Execute(); err != nil {
 		logger.Failuref("%v", err)
 		os.Exit(1)
 	}
 }
 
-func configureKubeconfig() {
-	switch {
-	case len(rootArgs.kubeconfig) > 0:
-	case len(os.Getenv("KUBECONFIG")) > 0:
-		rootArgs.kubeconfig = os.Getenv("KUBECONFIG")
-	default:
-		if home := homeDir(); len(home) > 0 {
-			rootArgs.kubeconfig = filepath.Join(home, ".kube", "config")
-		}
-	}
-}
-
 func configureDefaultNamespace() {
+	*kubeconfigArgs.Namespace = rootArgs.defaults.Namespace
 	fromEnv := os.Getenv("FLUX_SYSTEM_NAMESPACE")
-	if fromEnv != "" && rootArgs.namespace == rootArgs.defaults.Namespace {
-		rootArgs.namespace = fromEnv
+	if fromEnv != "" {
```

Comment:
This is not ok, please restore the previous behaviour, we only look for `FLUX_SYSTEM_NAMESPACE ` if the namespace arg is different to the default `flux-system`.

Reply:
The behaviour is that a non default value takes precedence over the env var.

Reply:
The precedence is still the same. It uses the cli-arg, then the variable, then the default.
We set `kubeconfigArgs.Namespace` to default, then - if set - to the env variable. The `AddFlags()` method will then use whatever is in `kubeconfigArgs.Namespace` as the default for the flag.

Reply:
This should use the `bar` namespace:
```
export FLUX_SYSTEM_NAMESPACE=foo
flux get all -n bar
```

This should use the `foo` namespace:
```
export FLUX_SYSTEM_NAMESPACE=foo
flux get all -n flux-system
```

This should use the `foo` namespace:
```
export FLUX_SYSTEM_NAMESPACE=foo
flux get all
```

Reply:
That's not the current behavior though:
```
~/projects$ flux version --client                                                                                                                                  schiff-controlplane-reftmdc.reftmdc.bn
flux: v0.24.0
~/projects$ export FLUX_SYSTEM_NAMESPACE=gitops; flux reconcile source git -n flux-system deployment                                               schiff-controlplane-reftmdc.reftmdc.bn
✗ gitrepositories.source.toolkit.fluxcd.io "deployment" not found
~/projects$ flux reconcile source git -n gitops deployment                                                                                         schiff-controlplane-reftmdc.reftmdc.bn
► annotating GitRepository deployment in gitops namespace
```